app/assets/javascripts/documents.js.coffee

```diff
@@ -19,6 +19,24 @@
 # TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.~
 
 $ ->
+
+  setDocumentTableInterval = null
+
+  $(document).on 'click', '.report-request', ->
+    setDocumentTableInterval = setInterval refreshDocumentTable, 5000
+
+  clearDocumentTableInterval = (complete) ->
+    if complete == true
+      clearInterval setDocumentTableInterval
+
+  refreshDocumentTable = ->
```

Comment:
A few things here.
Could you run the refresh on page load if there are documents that are still processing?
Also could you add a more specific selector by adding a class to the spans in the helper?
And then could you remove the `on 'click', '.report-request'` part, add an export to the refresh function, and add a call to the refresh to `views/reports/create.js.coffee` if the report was valid?

Reply:
I'll leave it like this for now because I like the way it works.